libraries/SITL/SIM_VectorNav.cpp

```diff
@@ -0,0 +1,229 @@
+    float GPS1velNED[3];
+    float GPS1DOP[7];
+    uint8_t numGPS2Sats;
+    uint8_t GPS2Fix;
+    float GPS2DOP[7];
+};
+
+#define VN_PKT1_HEADER { 0xfa, 0x35, 0x03, 0x00, 0x2c, 0x0f, 0x47, 0x01, 0x13, 0x06 }
+#define VN_PKT2_HEADER { 0xfa, 0x4e, 0x02, 0x00, 0x10, 0x00, 0xb8, 0x20, 0x18, 0x20 }
+
+/*
+  get timeval using simulation time
+ */
+static void simulation_timeval(struct timeval *tv)
+{
+    uint64_t now = AP_HAL::micros64();
+    static uint64_t first_usec;
+    static struct timeval first_tv;
+    if (first_usec == 0) {
+        first_usec = now;
+        first_tv.tv_sec = AP::sitl()->start_time_UTC;
+    }
+    *tv = first_tv;
+    tv->tv_sec += now / 1000000ULL;
+    uint64_t new_usec = tv->tv_usec + (now % 1000000ULL);
+    tv->tv_sec += new_usec / 1000000ULL;
+    tv->tv_usec = new_usec % 1000000ULL;
+}
+
+void VectorNav::send_packet1(void)
+{
+    const auto &fdm = _sitl->state;
+
+    struct VN_packet1 pkt {};
+
+    struct timeval tv;
+    simulation_timeval(&tv);
+
+    if (start_us == 0) {
+        start_us = tv.tv_usec * 1000;
+    }
+
+    pkt.timeStartup = start_us;
+    pkt.timeGPS = tv.tv_usec * 1000;
+    pkt.uncompAccel[0] = fdm.xAccel;
+    pkt.uncompAccel[1] = fdm.yAccel;
+    pkt.uncompAccel[2] = fdm.zAccel;
+    const float gyro_noise = 0.05;
+    pkt.uncompAngRate[0] = radians(fdm.rollRate + gyro_noise * rand_float());
+    pkt.uncompAngRate[1] = radians(fdm.pitchRate + gyro_noise * rand_float());
+    pkt.uncompAngRate[2] = radians(fdm.yawRate + gyro_noise * rand_float());
+
+    float sigma, delta, theta;
+    AP_Baro::SimpleAtmosphere(fdm.altitude * 0.001f, sigma, delta, theta);
+    pkt.pressure = SSL_AIR_PRESSURE * delta * 0.001 + rand_float() * 0.01;
+
+    pkt.mag[0] = fdm.bodyMagField.x*0.001;
+    pkt.mag[1] = fdm.bodyMagField.y*0.001;
+    pkt.mag[2] = fdm.bodyMagField.z*0.001;
+
+    pkt.accel[0] = fdm.xAccel;
+    pkt.accel[1] = fdm.yAccel;
+    pkt.accel[2] = fdm.zAccel;
+    pkt.gyro[0] = radians(fdm.rollRate + rand_float() * gyro_noise);
+    pkt.gyro[1] = radians(fdm.pitchRate + rand_float() * gyro_noise);
+    pkt.gyro[2] = radians(fdm.yawRate + rand_float() * gyro_noise);
+    
+    // pkt.sensSat = ???
+    // pkt.AHRSStatus = ???
+
+    pkt.ypr[0] = fdm.yawDeg;
+    pkt.ypr[1] = fdm.pitchDeg;
+    pkt.ypr[2] = fdm.rollDeg;
+
+    pkt.quaternion[0] = fdm.quaternion.q2;
+    pkt.quaternion[1] = fdm.quaternion.q3;
+    pkt.quaternion[2] = fdm.quaternion.q4;
+    pkt.quaternion[3] = fdm.quaternion.q1;
+
+    pkt.linAccBody[0] = fdm.xAccel;
+    pkt.linAccBody[1] = fdm.yAccel;
+    pkt.linAccBody[2] = fdm.zAccel;
+
+    // pkt.yprU[3] = attitude uncertainty
+
+    // pkt.INSStatus = ??
+
+    pkt.positionLLA[0] = fdm.latitude;
+    pkt.positionLLA[1] = fdm.longitude;
+    pkt.positionLLA[2] = fdm.altitude;
+    pkt.velNED[0] = fdm.speedN;
+    pkt.velNED[1] = fdm.speedE;
+    pkt.velNED[2] = fdm.speedD;
+    pkt.posU = 0.5;
+    pkt.velU = 0.25;
+
+    const uint8_t header[] VN_PKT1_HEADER;
```

Comment:
You know you could've defined this up the top as a `static const uint8_t pkt1_header[]` and done away with `VN_PKT1_HEADER` :-P

Reply:
done

Reply:
... not as far as I can see...